server/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -149,12 +149,11 @@ public boolean enoughShardsActive(final ClusterState clusterState, final String.
                 return false;
             }
             ActiveShardCount waitForActiveShards = this;
-            if (waitForActiveShards == ActiveShardCount.DEFAULT) {
+            if (waitForActiveShards == ActiveShardCount.DEFAULT && SETTING_WAIT_FOR_ACTIVE_SHARDS.exists(indexMetadata.getSettings())) {
                 waitForActiveShards = SETTING_WAIT_FOR_ACTIVE_SHARDS.get(indexMetadata.getSettings());
```

Comment:
This was always resolving `waitForActiveShards` to non default one. A a result branch below was a dead code.

Reply:
Maybe we should assign a special meaning to `1` instead of comparing to default.